src/SprykerEco/Zed/AfterPay/Communication/Plugin/Checkout/AfterPaySaveOrderPlugin.php

```diff
@@ -34,6 +36,17 @@ class AfterPaySaveOrderPlugin extends AbstractPlugin implements CheckoutDoSaveOr
      */
     public function saveOrder(QuoteTransfer $quoteTransfer, SaveOrderTransfer $saveOrderTransfer): void
     {
-        $this->getFacade()->saveOrderPayment($quoteTransfer, $saveOrderTransfer);
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== SharedAfterPayConfig::PROVIDER_NAME) {
```

Comment:
* Plugin doesn't contain business logic and just delegate to the underlying Module API (Facade, Client, or Service).
* Move the condition under the hood of the `saveOrderPayment()` method.